### PR TITLE
zalgo: init at unstable-2020-08-26

### PR DESCRIPTION
**Pull request by djanatyn**
###### Motivation for this change
I need a way to generate text that looks haunted on NixOS this month.
T͕h̢̤̲í̙̞s t̡o̳̙o̳͢ļ̭̦̘ w̛̺il̞l s͖̗úf͓̮͜f̫̟͞i͉̟̠cẹ̩̱͝.

Z̛̼̩̤̲̘͜͞Ą̟͈̳̥̙ͅL̵͈̪̲̣̺̮G̢̪̞̦̱͖͓̳O̮̜̬̳̻̫͜͞.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by djanatyn** [comment]
This is a new project that doesn't have a traditional version scheme (no semantic versioning). I'm not sure if using the git hash in the `version` attribute is appropriate?

File: pkgs/tools/misc/zalgo/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    rev = version;
+    sha256 = "00q56yvfcj2f89wllrckvizihivqmd6l77nihb52ffqd99rdd24w";
+  };
+
+  installPhase = ''
+    install -Dt $out/bin zalgo
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Read stdin and corrupt it with combining diacritics.";
```
Comment (IvarWithoutBones) [inline pkgs/tools/misc/zalgo/default.nix:19]: `meta.description` should not end with a period, according to the [contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes).

File: pkgs/tools/misc/zalgo/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zalgo";
+  version = "6aa1f66";
```
Comment (IvarWithoutBones) [inline pkgs/tools/misc/zalgo/default.nix:5]: From the [nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#sec-package-naming): `If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".`

```suggestion
  pname = "zalgo-unstable";
  version = "2020-08-26";
```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
A few minor issues, otherwise LGTM :+1:

File: pkgs/tools/misc/zalgo/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zalgo-unstable";
+  version = "2020-08-26";
+
+  src = fetchFromGitHub {
+    owner = "lunasorcery";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/zalgo/default.nix:9]:
```suggestion
    repo = "zalgo";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by djanatyn** [comment]
Updated!

File: pkgs/tools/misc/zalgo/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zalgo-unstable";
```
Comment (sikmir) [inline pkgs/tools/misc/zalgo/default.nix:4]: Move `unstable` prefix to version.

File: pkgs/tools/misc/zalgo/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  version = "2020-08-26";
+
+  src = fetchFromGitHub {
+    owner = "lunasorcery";
+    repo = "zalgo";
+    rev = version;
```
Comment (sikmir) [inline pkgs/tools/misc/zalgo/default.nix:10]: How it works for you? There is no tags in the repo.

File: pkgs/tools/misc/zalgo/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "00q56yvfcj2f89wllrckvizihivqmd6l77nihb52ffqd99rdd24w";
+  };
+
+  installPhase = ''
+    install -Dt $out/bin zalgo
```
Comment (sikmir) [inline pkgs/tools/misc/zalgo/default.nix:15]:
```suggestion
    install -Dm755 zalgo -t $out/bin
```

File: pkgs/tools/misc/zalgo/default.nix (hunk 5 above)
Comment (djanatyn) [reply pkgs/tools/misc/zalgo/default.nix]: Sorry, I was just running `nix-build -A zalgo` without updating the source!

I misunderstood one of the earlier review comments. I think dedf0de addresses this.

Comment (djanatyn) [reply pkgs/tools/misc/zalgo/default.nix]: I looked at some other examples and updated again, which I think is more in line with the established convention and notes in the nixpkgs manual.

File: pkgs/tools/misc/zalgo/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  name = "zalgo-unstable-2020-08-26";
```
Comment (IvarWithoutBones) [inline pkgs/tools/misc/zalgo/default.nix:4]: Using `name` is deprecated.
```suggestion
  pname = "zalgo-unstable";
  version = "2020-08-26";
```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
Commits should be squashed as well

Other than this, LGTM :)

File: pkgs/tools/misc/zalgo/default.nix [hunk 8]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zalgo-unstable";
+  version = "2020-08-26";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/zalgo/default.nix:5]:
```suggestion
  pname = "zalgo";
  version = "unstable-2020-08-26";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Can you please change the PR title and the commit message to:
```
zalgo: init at unstable-2020-08-26
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101428` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zalgo</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101428` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zalgo</li>
  </ul>
</details>

**Comment by djanatyn** [comment]
> Can you please change the PR title and the commit message to:
> 
> ```
> zalgo: init at unstable-2020-08-26
> ```

Done, pushed 383bd9f.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101428` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zalgo</li>
  </ul>
</details>

**Pull request merged**